plots/chart.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* global Plotly, document, generatedResults */
+
+const IGNORED_METRICS = ['Navigation Start'];
+
+let elementId = 1;
+
+function generateBoxPlotChartPerMetric() {
+  for (const metric in generatedResults) {
+    if (IGNORED_METRICS.indexOf(metric) !== -1) {
+      continue;
+    }
+    generateBoxPlotChartByBatch({metric, type: 'timing'});
+  }
+
+  function generateBoxPlotChartByBatch({metric, type}) {
+    const width = 30;
+    for (let i = 0; i < generatedResults[metric].length; i += width) {
+      generateBoxPlotChart({
+        title: metric,
+        data: generatedResults[metric].slice(i, i + width),
+        type: type
+      });
+    }
+  }
+
+  function generateBoxPlotChart({title, data, type}) {
+    data = data
+      .map(siteResult => {
+        return {
+          x: siteResult.metrics.map(m => {
+            if (!m) {
+              return;
+            }
+            return m[type];
+          }),
+          type: 'box',
+          name: siteResult.site,
+          boxpoints: 'all',
+          jitter: 0.9,
+          pointpos: -2,
+          hoverinfo: 'x+name'
+        };
+      })
+      .reverse();
+
+    const layout = {title: title + ' ' + type};
+    Plotly.newPlot(createChartElement(1000), data, layout);
```

Comment:
since we flood all these plots at once i see my machine stall for 5s while plotly goes to work.

here's an incremental raf-scheduler for them: https://github.com/GoogleChrome/lighthouse/commit/d5405e72a6904b8ff9d17ab0571e1b108a264c28